### DISCUSSION TITLE
How to Pan and Zoom Image, add Markers on it and move markers as it is pan and zoomed?

### DISCUSSION THREAD

**Discussion by gpbaculio** [software-mansion/react-native-reanimated#5225] [Q&A]
I am looking at implementing this in skia or reanimated, I am just confused how to do it, I have this Image
![Rectangle 41](https://github.com/software-mansion/react-native-reanimated/assets/17801912/8c70a182-7e4b-4546-a599-b3f0a2486a73)
and in React Native I need to add circle markers as you see in this screenshot:
![Screenshot 2023-10-13 at 10 36 47 AM](https://github.com/software-mansion/react-native-reanimated/assets/17801912/509634b0-cd26-4302-b7be-3f482ea938f1)
I have implemented pan and zoom before on Image but I am confused how to add those markers and render a component on press, and make those markers move as it is zoom and pan

**Comment by MatiPl01** [comment]
Hey!

I know that this discussion is old but [this](https://github.com/enzomanuelmangano/react-native-skia-gesture) library may be helpful.

You cannot just simply recognize press gesture on elements rendered on the Skia Canvas. Another way to do this is just rendering all of circles and tooltips with React Native over the canvas. You would be able to easily control press gestures then.
To properly move these markers with the canvas, you would have to apply transforms to the parent component of the canvas and all marker components.
